content/advent-2015/generate-free-tls-certs-with-lego.md

```diff
@@ -0,0 +1,204 @@
++++
+author = ["Matt Holt"]
+date = "2015-12-15T08:00:00-07:00"
+series = ["Advent 2015"]
+title = "Generate and Use Free TLS Certificates with Lego"
++++
+
+If your Go program uses the transport layer of the network at all&mdash;whether to serve static files, an API, or something else over the wire&mdash;you should be encrypting connections using TLS. Hopefully this is obvious by now. But developers still don't do it. Why?
+
+Because TLS (formerly "SSL") certificates cost money and require manual labor to obtain, install, and maintain. Besides, there's no reason to encrypt unless you collect or send sensitive data, right? **Wrong.** Encryption not only prevents eavesdropping and surveillance, it also protects packets from being modified in flight&mdash;modifications that could break your API or track your users. Essentially, TLS adds a layer of privacy and integrity to your application.
+
+This post will guide you through a free, easy way to obtain real, trusted TLS certificates using Go. Thanks to the efforts of the ISRG and, in particular, [Let's Encrypt](https://letsencrypt.org), the ACME protocol makes it possible to do this. Sebastian Erhart has done an excellent job building an ACME client in Go called [lego](https://github.com/xenolf/lego) that we can use to get free, valid TLS certificates in seconds. The technique shown here is very similar to what [Caddy](https://caddyserver.com) does to serve your sites over HTTPS by default.
```

Comment:
You spell out "Internet Security Research Group" and link it below, but it'd be nice to do one of those things up here at the first reference. Otherwise I'm wondering what ISRG for a few paragraphs (or turn to Google).